src/Fantasma/Internal/Storage/EntityFramework/EntityFrameworkStorage.cs

```diff
@@ -0,0 +1,141 @@
+
+        if (await db.Jobs.FindAsync(job.Id) != null)
+        {
+            return;
+        }
+
+        db.Jobs.Add(
+            new FantasmaJob
+                {
+                    Id = job.Id,
+                    ClrType = job.Data.GetType().AssemblyQualifiedName ?? throw new InvalidOperationException(),
+                    Data = JsonConvert.SerializeObject(job.Data),
+                    ScheduledAt = job.ScheduledAt.UtcDateTime,
+                    Status = JobStatus.Scheduled,
+                    Cron = job.Cron,
+                });
+
+        await db.SaveChangesAsync();
+    }
+
+    public async Task Remove(Job job)
+    {
+        using var scope = _provider.CreateScope();
+        using var db = scope.ServiceProvider.GetRequiredService<IFantasmaDatabase>();
+
+        var dbJob = await db.Jobs.FindAsync(job.Id);
+        if (dbJob != null)
+        {
+            db.Remove(dbJob);
+            await db.SaveChangesAsync();
+        }
+    }
+
+    public async Task Update(Job job)
+    {
+        using var scope = _provider.CreateScope();
+        using var db = scope.ServiceProvider.GetRequiredService<IFantasmaDatabase>();
+
+        var dbJob = await db.Jobs.FindAsync(job.Id);
+        if (dbJob != null)
+        {
+            dbJob.Status = job.Status;
+            dbJob.ScheduledAt = job.ScheduledAt.UtcDateTime;
+            await db.SaveChangesAsync();
+        }
+    }
+
+    public Task<Job?> GetNextJob()
+    {
+        using var scope = _provider.CreateScope();
+        using var db = scope.ServiceProvider.GetRequiredService<IFantasmaDatabase>();
+
+        var now = _time.GetUtcNow();
+        var nextJob = db.Jobs.Where(x => x.ScheduledAt < now).AsEnumerable().MinBy(x => x.ScheduledAt);
+        if (nextJob == null)
+        {
+            return Task.FromResult<Job?>(null);
+        }
+
+        var type = Type.GetType(nextJob.ClrType);
+        if (type == null)
+        {
+            _logger.LogError("Could not get CLR type for job");
+            return Task.FromResult<Job?>(null);
+        }
+
+        if (JsonConvert.DeserializeObject(nextJob.Data, type) is not IJobData data)
+        {
+            _logger.LogError("Could not deserialize job data");
+            return Task.FromResult<Job?>(null);
+        }
+
+        return Task.FromResult<Job?>(
+            new Job
+                {
+                    Id = nextJob.Id,
+                    ScheduledAt = nextJob.ScheduledAt,
+                    Status = JobStatus.Scheduled,
+                    Data = data,
+                    Cron = nextJob.Cron,
+                });
+    }
+
+    public async Task Release(Job job)
+    {
+        using var scope = _provider.CreateScope();
+        using var db = scope.ServiceProvider.GetRequiredService<IFantasmaDatabase>();
+
+        var dbJob = await db.Jobs.FindAsync(job.Id);
+        if (dbJob != null)
+        {
+            db.Remove(dbJob);
+            await db.SaveChangesAsync();
+        }
```

Comment:
nit: can use execute delete to avoid loading the dbJob
```suggestion
        context.Jobs.Where(j => j.Id == job.Id).ExecuteDelete();
```

Reply:
The ExecuteDelete extension method doesn't seem to be available for me. Does that reside in another NuGet package?

Reply:
Ah, need to reference the `microsoft.entityframeworkcore.relational` package for that to work.